build_tools/scripts/update_submodules.sh

```diff
@@ -37,8 +37,9 @@ if [ -n "$git_status" ]; then
   exit 1
 fi
 
-echo "Pulling latest from remote"
-git pull
+upstream_url="https://github.com/google/iree.git"
+echo "Pulling latest from remote ${upstream_url?}"
+git pull ${upstream_url}
```

Comment:
I was wondering if we should just do `git pull ${upstream_url} --rebase` ? That works better on fork. I dont see merge commits (but I have to do git push --force sometimes).
I can't test this on the master branch, but seems reasonable.

Reply:
Seems reasonable to me. However reading the man page:

> This is a potentially dangerous mode of operation. It rewrites history, which does not bode well when you published that history already. Do not use this option unless you have read git-rebase(1) carefully.

So maybe not.

I was actually thinking about whether we should have this create a new branch instead of committing to master. We were just pushing straight to master for these, since we have to do it all the time and we didn't have any presubmits that would test it anyway, but now that we've actually got presubmits it might be better to do these as PRs just like everything else.

Let's keep this as an incremental improvement though

Reply:
Agreed. Rebase doesnt work that well for whatever reason. Submitted this one.